### ISSUE TITLE
Download progress not working

### ISSUE THREAD

**Issue by Vingtoft** [daltoniam/SwiftHTTP#212]
Im having some trouble getting the download/upload progress to work. 
At this moment the progress is always 0.0.

This is my code:
```
    // Upload file to server.
    func uploadFileToServer(path_to_file:String) {
        let fileUrl = NSURL(fileURLWithPath: path_to_file)
        do {
            let opt = try HTTP.POST(url, parameters: ["file": Upload(fileUrl: fileUrl)])
            opt.progress = { progress in
                Swift.print("Progess is: \(progress)")
            }
            opt.start { response in
                Swift.print("Response: \(response)")
            }
        } catch let error {
            Swift.print("got an error creating the request: \(error)")
        }
    }
```

The output is:
```
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Progess is: 0.0
Response: SwiftHTTP.Response
```

Any ideas how I can get it to work?

**Comment by daltoniam** [comment]
Interesting. It appears the progress closure is getting called multiple, just not getting an updated value. You might want to check here:
https://github.com/daltoniam/SwiftHTTP/blob/master/Source/Operation.swift#L447

Try printing and see if the URLSession delegate is reporting the request's upload progress (expect bytes and total bytes sent).

**Comment by serendipityapps** [comment]
Bug in Operation.swift - func progressHandler(response: Response, expectedLength: Int64, currentLength: Int64)
- Calculation of 'var slice' is integer/integer which always produces integer not float 

Fix by explicitly changing to Float 
i.e.
let slice = Float(1.0)/Float(expectedLength)
handler(slice*Float(currentLength))

Fix and test included in Pull request #215

**Comment by daltoniam** [comment]
#215 has been merged.
